tools/check-test-fix.sh

```diff
@@ -0,0 +1,75 @@
+#! /usr/bin/env bash
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2019, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# This will re-run a LISA test against data referenced by a bisector report.
+#
+# Usage: check-test-fix.sh EXEKALL_TEST_ID BISECTOR_REPORT [BISECTOR_OPTION ...]
+# example:
+# check-test-fix.sh 'RampDown:test_task_placement' TC2.report.yml.gz -oiterations=1-10 --only behaviour
+
+exekall_test=$1
+shift
+bisector_report=$1
+shift
+bisector_options=$@
+
+exekall_load_type='*TestBundle'
+
+bisector_logs_folder=$(readlink -f "${bisector_report}.artifacts")
+extracted_artifacts_folder=$bisector_logs_folder/extracted_exekall_artifacts
+exekall_reference_db=$bisector_logs_folder/VALUE_DB.pickle.xz
+exekall_new_artifacts=$bisector_logs_folder/new_exekall_artifacts
+exekall_merged_artifact=$bisector_logs_folder/merged_new_exekall_artifacts
+exekall_compare_log=${bisector_report}.comparison
+
+set -x
+
+# Get all the artifact folders containing the traces
+rm -rf "$bisector_logs_folder" 2>/dev/null
+bisector report "$bisector_report"		\
+    -oexport-logs="$bisector_logs_folder"	\
+    -oexport-db="$exekall_reference_db"		\
+    -odownload					\
+    ${bisector_options[@]}
+
+# Uncompress all artifact archives
+mkdir -p "$extracted_artifacts_folder" &&
+find  "$bisector_logs_folder" -name '*.tar.xz' -exec \
+    tar -C "$extracted_artifacts_folder" -xvf '{}' ';'
+
+# Run the test against all DBs that we extracted.
+# Running on one DB at a time instead of merging first allows constant memory
+# usage, instead of having to accumulate all results.
+find "$extracted_artifacts_folder" -name 'VALUE_DB.pickle.xz' -print0	\
+    | xargs -0 -n1 -I '{}'	\
+	exekall run "$LISA_HOME/lisa/tests/" \
```

Comment:
This step really takes forever on large data sets :/. One possibility could be to parallelize this a bit. I tried using GNU parallel and got the following results (10 iterations of all `test_task_placement` on TC2):

```
$ time find TC2.report.yml.gz.artifacts/extracted_exekall_artifacts \
    -name VALUE_DB.pickle.xz -print0 | xargs -0 -n1 -I '{}' \
       exekall run "$LISA_HOME/lisa/tests/" --load-db "{}" \
       --artifact-root TC2.report.yml.gz.artifacts/new_exekall_artifacts \
       --select '*:test_task_placement' --load-type '*TestBundle

real	2m27.946s
user	2m52.986s
sys	1m12.864s
```
And with parallel
```
$ time find TC2.report.yml.gz.artifacts/extracted_exekall_artifacts \
    -name VALUE_DB.pickle.xz | parallel -I '{}' --max-args 1 \
       exekall run "$LISA_HOME/lisa/tests/" --load-db "{}" \
       --artifact-root TC2.report.yml.gz.artifacts/new_exekall_artifacts \
       --select '*:test_task_placement' --load-type '*TestBundle'

real	0m24.793s
user	3m53.745s
sys	1m9.134s
```

`parallel` is in the ubuntu repo, so perhaps it's OK to add it as a dependency ?

Reply:
got it working with `xargs -P`. Also made uncompressing parallel, with as many processes as possible